sklearn/utils/estimator_checks.py

```diff
@@ -252,6 +241,21 @@ def _check_transformer(name, Transformer, X, y):
             assert_raises(ValueError, transformer.transform, X.T)
 
 
+def check_fit_score_takes_y(name, Estimator):
+    # check that all estimators accept an optional y
+    # in fit and score so they can be used in pipelines
+    rnd = np.random.RandomState(0)
+    X = rnd.uniform(size=(10, 3))
+    y = (X[:, 0] * 4).astype(np.int)
+    y = multioutput_estimator_convert_y_2d(name, y)
+    with warnings.catch_warnings(record=True):
+        estimator = Estimator()
```

Comment:
I thought about adding a test for ``partial_fit``, but I'm not sure about it. The pipelines need to look quite a bit different to support ``partial_fit``, and there is no pressing reason currently to enforce taking a ``y`` in the API....